airflow/operators/bash.py

```diff
@@ -157,14 +163,37 @@ def pre_exec():
 
             self.log.info('Running command: %s', self.bash_command)
 
-            self.sub_process = Popen(  # pylint: disable=subprocess-popen-preexec-fn
-                ['bash', "-c", self.bash_command],
-                stdout=PIPE,
-                stderr=STDOUT,
-                cwd=tmp_dir,
-                env=env,
-                preexec_fn=pre_exec,
-            )
+            if isinstance(self.bash_command) == str and not _is_bash_script(self.bash_command):
+                warnings.warn(
+                    "Warning: Using a string in the BashOperator leaves your system open to bash injection "
+                    "attacks via escape strings. Please use a list[str] instead."
+                )
+                self.sub_process = Popen(  # pylint: disable=subprocess-popen-preexec-fn
+                    ['bash', "-c", self.bash_command],
+                    stdout=PIPE,
+                    stderr=STDOUT,
+                    cwd=tmp_dir,
+                    env=env,
+                    preexec_fn=pre_exec,
+                )
+            elif isinstance(self.bash_command) == list:
+                if (
+                    len(self.bash_command) >= 2
+                    and self.bash_command[0] == "bash"
+                    and self.bash_command[1] == "-c"
+                ):
```

Comment:
Can't do multi line suggest on mobile app but

```bash_command[0:2] == ["bash", "-c"]```

Reply:
This is good, they should get hit with a warning that is functionally equivalent and still leads to evaluation of bash entities.

Reply:
Yes, running bash is the point of the operator - it's called a BashOperetor, not ExecCommandOperator ;)